catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.kt

```diff
@@ -0,0 +1,1009 @@
+        listView = view.findViewById(android.R.id.list)
+        scriptFinder = view.findViewById(R.id.findview)
+
+        scriptFinder?.setOnResultFoundListener(object : OnResultFoundListener {
+            override fun onResultFound(
+                sceneIndex: Int,
+                spriteIndex: Int,
+                brickIndex: Int,
+                totalResults: Int,
+                textView: TextView?
+            ) {
+                val currentProject =
+                    projectManager.currentProject
+                val currentScene =
+                    currentProject.sceneList[sceneIndex]
+                val currentSprite =
+                    currentScene.spriteList[spriteIndex]
+                textView?.text = createActionBarTitle(
+                    currentProject,
+                    currentScene,
+                    currentSprite
+                )
+                projectManager.setCurrentSceneAndSprite(
+                    currentScene.name,
+                    currentSprite.name
+                )
+                adapter.updateItems(currentSprite)
+                adapter.notifyDataSetChanged()
+                listView?.smoothScrollToPosition(brickIndex)
+                highlightBrickAtIndex(brickIndex)
+                hideKeyboard()
+            }
+        })
+
+        val activity = activity as SpriteActivity?
+
+        scriptFinder?.setOnCloseListener(object : OnCloseListener {
+            override fun onClose() {
+                listView?.cancelHighlighting()
+                finishActionMode()
+                if (activity != null && !activity.isFinishing) {
+                    activity.setCurrentSceneAndSprite(
+                        projectManager.currentlyEditedScene,
+                        projectManager.currentSprite
+                    )
+                    activity.supportActionBar?.title = activity.createActionBarTitle()
+                    activity.addTabs()
+                }
+                activity?.findViewById<View>(R.id.toolbar)?.visibility = View.VISIBLE
+            }
+        })
+
+        scriptFinder?.setOnOpenListener(object : OnOpenListener {
+            override fun onOpen() {
+                activity?.removeTabs()
+                activity?.findViewById<View>(R.id.toolbar)?.visibility = View.GONE
+            }
+        })
+
+        setHasOptionsMenu(true)
+        SnackbarUtil.showHintSnackbar(getActivity(), R.string.hint_scripts)
+        return view
+    }
+
+    fun createActionBarTitle(
+        currentProject: Project,
+        currentScene: Scene,
+        currentSprite: Sprite
+    ): String {
+        return if (currentProject.sceneList.size == 1) {
+            currentSprite.name
+        } else {
+            currentScene.name + ": " + currentSprite.name
+        }
+    }
+
+    private fun highlightBrickAtIndex(index: Int) {
+        listView?.getBrickPositionsToHighlight()?.clear()
+        listView?.getBrickPositionsToHighlight()?.add(index)
```

Comment:
```suggestion
        listView?.brickPositionsToHighlight?.add(index)
```

Reply:
brickPositionsToHighlight is currently public and while refactoring BrickListView.kt to kotlin in https://github.com/Catrobat/Catroid/pull/4246 I keep it the same, its giving the following error.

![image](https://user-images.githubusercontent.com/46495964/125186533-74cc4980-e248-11eb-880f-da526a03fd06.png)

Reply:
yeah but in kotlin, you don't need encapsulation, just remove the getter and declare the setter as private if not needed.
see [Getters and setters﻿](https://kotlinlang.org/docs/properties.html#getters-and-setters)

Reply:
So these properties could be public and there isn't any need for setter in these so don't need to define it explicitly, please correct me if I'm wrong !
![image](https://user-images.githubusercontent.com/46495964/125203909-542be000-e298-11eb-97ae-a725dbfd1e28.png)